File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length ;
        int i = 0 ;
        int left = 0 , right = n -1 ;

        while (i <= right)
        {
            if (nums[i] == 0)
            {
                int temp = nums[i] ;
                nums[i] = nums[left] ;
                nums[left] = temp ; 
                left++ ;
                i++ ;
            }
            else if (nums[i] == 2)
            {
                int temp1 = nums[i] ;
                nums[i] = nums[right] ;
                nums[right] = temp1 ;
                right -- ;
            }
            else
            {
                i ++ ;
            }
        }
        
    }
}

File: 75-sort-colors/sort-colors.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        l = 0 
        r = len(nums) - 1
        i = 0

        while l <= r:

            if nums[l] == 0:
                nums[l] , nums[i] = nums[i] , nums[l]
                i += 1
                l += 1

            elif nums[l] == 2:
                nums[l] , nums[r] = nums[r] , nums[l]
                r -= 1
            
            else:
                l += 1



            
        